File: 4332-check-good-integer/check-good-integer.py
class Solution:
    def checkGoodInteger(self, n: int) -> bool:
        squreSum = digitSum = 0

        while n:
            r = n % 10
            digitSum += r
            squreSum += r * r
            n //= 10

        return squreSum - digitSum >= 50

File: 4332-check-good-integer/check-good-integer.go
func checkGoodInteger(n int) bool {
	digitSum := 0
	squreSum := 0

	for n > 0 {
		r := n % 10
		digitSum += r
		squreSum += r * r
		n /= 10
	}

	return squreSum-digitSum >= 50
}